cxx4opencl/diff2openclc.txt

```diff
@@ -0,0 +1,120 @@
+=== Difference to OpenCL C
+
+{cpp} for OpenCL targets backwards compatibility to OpenCL C
+in the majority of features. However, it is a different
+language that is derived from {cpp} and therefore it inherits
+its fundamental design principles. Hence {cpp} for OpenCL will
+deviate from OpenCL C in the same areas where {cpp} deviates
+from C.
+
+In this section some of the most common areas where it is
+expected to behave differently are described.
+
+==== Implicit Conversions
+
+{cpp} is much more strict about conversions between types
+especially those that are performed implicitly by the compiler.
+For example it is not possible to convert non-`const` object
+to `const` explicitly. For details please refer to {cpp}17
+(`ISO/IEC 14882:2017`) `[conv]`.
+
+[source,c]
+----------
+void foo(){
+  const int* ptrconst;
+  int* ptr = ptrconst; // rejected in C++ for OpenCL
+}
+----------
+
+The same situation applies to narrowing conversions in
+initialization lists ({cpp}17 `[dcl.init.list]`). 
+
+[source,c]
+----------
+struct mytype {
+ int i;
+};
+void foo(uint par){
+  mytype var = {
+    .i = par
```

Comment:
It would help if you add a comment to highlight the issue.

Reply:
I would suggest adding an example such as `(int2)(4.2f, 3.14f)` as well.

Reply:
Btw I realized that this rule is just for the initialization list but I am not sure vector initialization falls under this category. For now I will avoid adding a vector example until I investigate this further.